packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts

```diff
@@ -55,7 +57,14 @@ export class PrometheusExporter implements MetricExporter {
    */
   constructor(config: ExporterConfig = {}, callback?: () => void) {
     this._logger = config.logger || new api.NoopLogger();
-    this._port = config.port || PrometheusExporter.DEFAULT_OPTIONS.port;
+    this._host =
+      config.host ||
+      process.env.OTEL_EXPORTER_PROMETHEUS_HOST ||
+      PrometheusExporter.DEFAULT_OPTIONS.host;
+    this._port =
+      config.port ||
+      Number(process.env.OTEL_EXPORTER_PROMETHEUS_PORT) ||
```

Comment:
what if Number will be converted to `NaN` ?

Reply:
The || default port is the saving grace. Notice I did not use ??

Reply:
In fact it will usually be NaN as it is. Do you prefer it be more explicit/thorough with the number checking?

Reply:
Having this in environment would be validated automatically with fallback to default. The whole idea for environment was to have one place for configuration with set of rules for validation defaults etc. This way you don't have to worry about the validation as it will be unified, less things to check etc. etc. The same we have with semantic conventions. If we want to change it / refactor etc. I think we should discuss it first on sig, otherwise we should keep it in environment.

Reply:
It would be interesting to export the parseNumber() functionality if the env vars continue to live here.

Reply:
it was marked as resolved but if we keep it here then case `Number(process.env.OTEL_EXPORTER_PROMETHEUS_PORT) ` and `NaN` is still valid

Reply:
`NaN || PrometheusExporter.DEFAULT_OPTIONS.port` might be good enough or am I being lazy?

Reply:
Should be fine i think

```
> Boolean(NaN)
false
```